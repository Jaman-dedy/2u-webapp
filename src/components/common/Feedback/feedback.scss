.feedbackWrapper {
    margin-top: 2rem;
    display: flex;
    height: 0%;
    animation: shake 0.82s cubic-bezier(0.36, 0.07, 0.19, 0.97) both;
    transform: translate3d(0, 0, 0);
    backface-visibility: hidden;
    perspective: 1000px;
    align-items: center;
    justify-content: center;
    width: 100%;
    .feedbackWrapper-content {
        text-align: center;
        .titleText {
            font-family: 'Montserrat';
            font-style: normal;
            font-weight: 600;
            margin-top: -23px;
            font-size: 30px;
            line-height: 37px;
            align-self: center;
            background-color: '#000000';
        }
        .messageText {
            font-family: 'Montserrat';
            font-style: normal;
            font-weight: 500;
            font-size: 18px;
            line-height: 22px;
        }
    }
}

@keyframes shake {
    10%,
    90% {
        transform: translate3d(-1px, 0, 0);
    }
    20%,
    80% {
        transform: translate3d(2px, 0, 0);
    }
    30%,
    50%,
    70% {
        transform: translate3d(-4px, 0, 0);
    }
    40%,
    60% {
        transform: translate3d(4px, 0, 0);
    }
}