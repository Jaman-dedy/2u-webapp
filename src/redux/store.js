import { applyMiddleware, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import apiMiddleware from '../middlewares/apiMiddleware';
import initialState from './initialState';
import rootReducer from './reducers';

const middleware = [apiMiddleware];

const store = createStore(
  combineReducers(rootReducer),
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

// expose store when run in Cypress
// if (window.Cypress) {
window.store = store;
// }

export default store;
