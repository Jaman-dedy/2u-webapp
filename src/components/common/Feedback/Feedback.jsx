/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import errorImage from 'assets/images/error.png';
import successImage from 'assets/images/confirm.png';
import './feedback.scss';

const Feedback = ({
  message,
  title,
  success,
  callbackFn,
  autoClose,
}) => {
  const [show, setShow] = useState(false);

  const dispatch = useDispatch();

  useEffect(() => {
    setShow(true);
    setTimeout(() => {
      if (autoClose) {
        setShow(false);
        if (callbackFn) {
          callbackFn()(dispatch);
        }
      }
    }, 3000);
  }, []);

  const close = () => {
    setShow(false);
    if (callbackFn) {
      callbackFn()(dispatch);
    }
  };

  return (
    <>
      {show ? (
        <div
          className="feedbackWrapper"
          onClick={() => {
            setShow(false);
            if (callbackFn) {
              callbackFn()(dispatch);
            }
          }}
        >
          <div className="feedbackWrapper-content">
            <div className="titleText">
              <h4>{global.translate(title).toUpperCase()}</h4>
              <p className="messageText">
                {global.translate(message)}
              </p>
            </div>
          </div>
          {!autoClose && (
            <button
              type="button"
              name="close"
              className="close cursor-pointer"
              onClick={close}
            >
              {global.translate('Close', 186)}
            </button>
          )}
        </div>
      ) : (
        <></>
      )}
    </>
  );
};
Feedback.propTypes = {
  message: PropTypes.string,
  title: PropTypes.string,
  success: PropTypes.bool,
  callbackFn: PropTypes.func,
  autoClose: PropTypes.bool,
};
Feedback.defaultProps = {
  title: 'SORRY',
  callbackFn: () => {},
  success: false,
  message: 'Invalid credential,please try again',
  autoClose: true,
};
export default Feedback;
