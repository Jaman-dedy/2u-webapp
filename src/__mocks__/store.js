import { createStore, applyMiddleware, combineReducers } from 'redux';
import rootReducer from 'redux/reducers';
import initialState from 'redux/initialState';
import apiMiddleware from 'middlewares/apiMiddleware';

const middleware = [apiMiddleware];

const mockStore = createStore(
  combineReducers(rootReducer),
  initialState,
  applyMiddleware(...middleware),
);

export default mockStore;
