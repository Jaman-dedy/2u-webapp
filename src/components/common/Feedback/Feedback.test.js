import React from 'react';
import { shallow } from 'enzyme';
import Feedback from './Feedback';
import store from '__mocks__/store';
import { Provider } from 'react-redux';

describe('Feedback', () => {
  it('should render correctly', () => {
    const props = {
      setShowWallet: jest.fn(),
      showWallet: false,
      data: {
        loading: false,
        data: [],
      },
    };
    const wrapper = shallow(
      <Provider store={store}>
        {' '}
        <Feedback {...props} />
      </Provider>,
    );
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
