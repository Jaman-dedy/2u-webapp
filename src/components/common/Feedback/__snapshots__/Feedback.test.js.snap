// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should render correctly 1`] = `ShallowWrapper {}`;
